### ISSUE TITLE
A few stations with bad detections that keep showing up

### ISSUE THREAD

**Issue by jsayers-bsc** [MotusDev/Motus-TO-DO#674]
Last week I was trying to help someone figure out why they had such a jump in bad detections all of a sudden. In their case it turned out to be a test tag near the station without a deployment combined with a surge in radio activity from one of the antennas. It's all good there now, but in digging around I noticed that many of the bad detections were being "detected" at many of the same stations, mostly in error.

Since then I've been noticing many of these same stations come up. If there are obvious bad detections at a particular site, there's a good chance one (or all) of these also detected the tag. Some of the tags are frequent fliers and clearly prone to being detected in error since they show up again and again at various sites. But others seem to be more like one offs. Though the plots show even more (not surprisingly) many of the problematic detections are making it through the filter onto the public detections tables. In most cases the bad detections spike for just a day or two.

Likely these have been spotted before, but I figured I'd just post them here. There are others that come up, but these seem to be the main ones I've been seeing. I'm curious what people's thoughts are. 

**Trunk Island Bermuda**
[Detections table](https://motus.org/data/receiverDeploymentDetections?n=1000&q=&f=1&o=0a1a2a3a4a5a&id=5520)
[Detections plot](https://motus.org/data/receiver/deployment/plot/?id=5520)

**Florida Panther NWR**
[Detections table](https://motus.org/data/receiverDeploymentDetections?id=4844)
[Detections plot](https://motus.org/data/receiver/deployment/plot/?id=4844)

**Hog Island**
[Detections table](https://motus.org/data/receiverDeploymentDetections?id=4852)
[Detections plot](https://motus.org/data/receiver/deployment/plot/?id=4852)

**Quintana**
[Detections table](https://motus.org/data/receiverDeploymentDetections?id=5149)
[Detections plot](https://motus.org/data/receiver/deployment/plot/?id=5149)

**Comment by Hussell** [comment]
See #610

**Comment by adamdsmith** [comment]
What filtering is being applied to these stations? @denislepage shared these data, and while there is an extremely large number of false detections at these times, relatively have `runLen` > 2.

![image](https://user-images.githubusercontent.com/3195855/75125924-92196b80-5685-11ea-8be8-a31057d1baa0.png)

**Comment by adamdsmith** [comment]
Not unexpected at all, but activity rates portend false positives. Will be interesting to see how general this is and how it relates to the whole codeset and a variety of common burst intervals. And since the tag finder seems to be [keeping a running log of noise every minute (by default)](https://github.com/MotusDev/Motus-TO-DO/issues/465), maybe we can leverage such a relationship to modify `pulses_to_confirm` to reduce false positives, or at least flag them with some probability of being a false positive...

![image](https://user-images.githubusercontent.com/3195855/75190565-9180e380-571e-11ea-902f-98461644e923.png)

**Comment by denislepage** [comment]
While we do provide runLen = 2 to users in the R package, we generally recommend they are ignored. In noisy periods (which we also provide in the R package now with something called run filters, I think), we also recommend that users increase the threshold for run length. It's a bit crude, but this is now described in the R book: https://motus.org/MotusRBook/dataCleaning.html#preliminary-filtering 

Part of what I am hoping for is to get better predictions for the occurrence of false positives, based on a better understanding of conditions where they occur.

**Comment by adamdsmith** [comment]
Yes, this should be one aspect we can simulate, though we'll want to use actual receiver data to inform aspects of the simulation (e.g., the frequency offset of noise pulses, whether noise bursts are truly random with regard to timing/frequency, etc.)

Related to those filters, how is `hourBin` calculated? That is, how can I either convert from `hourBin` back to an actual hour block (e.g., 2017-07-30 00:00), or join `alltags` records with the `activity` table?

**Comment by cjardine-bsc** [comment]
The hourbin of the run is the tsBegin divided by 3600

cast(b.tsBegin/3600 as integer) as hourbin

**Comment by adamdsmith** [comment]
Seeking confirmation of my understanding, as I didn't see it spelled out in the Motus R book (hoping I didn't miss it):
- `pulseCount` is total raw pulses detected in `hourBin`?
- `numTags` is total number of unique 'tags' matched with >= 1 run in `hourBin`?
- `numRuns` is total number of tag 'detections' (with `runLen` >= 2) in `hourBin`?

**Comment by denislepage** [comment]
I expect num tags does not include tags of runs lenght 1. Does are not generated by tag finder.

**Comment by cjardine-bsc** [comment]
Responses:

    pulseCount is total raw pulses detected in hourBin?

Yes, though not that this is sometimes less than the num hits, this confuses me. @leberrigan may have a better understanding. 

    numTags is total number of unique 'tags' matched with >= 1 run in hourBin?

Total number of unique tag IDs with run >=2 STARTING in that hourbin (if a run crosses more than one hour bin it is only considered in the first bin)

    numRuns is total number of tag 'detections' (with runLen >= 2) in hourBin?

the total number of runs STARTING in that hourbin (if a run crosses more than one hour bin it is only considered in the first bin)

So in the R book when it refers to "a high number of detections within an hour (e.g., >= 100)...", it's referring specifically to >= 100 numRuns within the hourBin?

Yes. 

And one more, how does motusFilter work on Lotek receivers? I find borderline detections on Lotek receivers difficult to evaluate...

It is essentially the same thing, just run on the data after it has been processed by the Lotek receiver, it does the same summary stats on the hits and assumes any hourbins with many short runs to be "noisy".

**Comment by adamdsmith** [comment]
Thanks for the clarification. Makes sense.

**Comment by jsayers-bsc** [comment]
This is all great stuff to eventually have in the [R book appendix](https://github.com/MotusDev/Motus-TO-DO/issues/676)!

**Comment by StuMackenzie** [comment]
As this all pertains to public outputs at some point - I think it might be wise for use to increase the pulse count filter in the public data outputs. Or make that customizable somehow so we can clean up noise on the fly.

**Comment by adamdsmith** [comment]
When attempting to link detections from `alltags` with noise info from the `activity` table, what is the recommended approach when there exist multiple batches for the same receiver/ant/hourBin with varying pulse/run data? It is reasonable to assume the entry with the most recent (presumably higher) batch ID is the most reasonable information?

In particular, I have a few cases where a given `batchID`/`hourBin` combo does not have an entry in the `activity` table, but there exists a different `batchID` for that same `hourBin`/`ant` combo to use. In that case, I simply make the join irrespective of `batchID`. However, by ignoring `batchID` I will also get 1-to-many joins for other detections with multiple `hourBin`/`ant` combos in the `activity` table. So, take the most recent? Pick the one with the most activity?

**Comment by cjardine-bsc** [comment]
Hi Adam, 

Thanks for the poke on this.

There are two things happening here. The first is the concept of duplicate (re-run) batches. This impacts batchID 338586. BatchID 854825 is a re-run of batch 338586 and therefore batch 338586 should be ignored.  We have half-way implemented a solution to this. We are adding a new table to the R package that will list all these deprecated batches so users can filter them out. We have added that table to the API but I believe @steffilazerte has not yet added it to the R package. I can follow up with her and ask about progress on that. Once you have a list of these deprecated batches it will make it easier for you to filter them out (ideally we'll also include a function that will delete these batches from your local dataset if you so wish). Note that if you do a full re-download of the data the duplicate batches should not appear in the runs or activity table. 

The second case involved hourbin 426209 in batches 854825 and 700087 this just happens to be the hour where one batch ends and the other begins. Therefore it should be included in both batches and the join would need to include batchID to account for this. 

So, in short, ignore batchID or always using the highest batchID for the join would not be a good solution. The better path forward is to make sure you have a list of deprecated batches and can remove those.

**Comment by adamdsmith** [comment]
Thanks Catherine. Tossing deprecated batches from the local database would be nice. Please let me know if there's an interim solution.

**Comment by cjardine-bsc** [comment]
[deprecatedBatchProj4.xlsx](https://github.com/MotusDev/Motus-TO-DO/files/4300408/deprecatedBatchProj4.xlsx)

Here is a list for you to work with while I follow up with @steffilazerte on automating this.

**Comment by denislepage** [comment]
That works, and another possible solution is to download a fresh copy of the entire project, since deprecated batches are excluded.

**Comment by adamdsmith** [comment]
Great, thanks!

**Comment by jsayers-bsc** [comment]
Here's another one just brought to our attention: [Deployment ID 4462]( https://motus.org/data/receiver/deployment/plot/?id=4462). 

Similar to the others, where there is a period of a few days where "detections" spike, then seemingly normal activity after.

This one is at a zoo and the detections table is often shared. Recently, this has resulted in a number of over excited local birders who mistakenly believed a seasonal rarity showed up. 

In this context, some way to address this on the publicly facing detections tables really seems important, even if it's just temporary. Along these lines, adding some sort of [disclaimer ](https://github.com/MotusDev/Motus-TO-DO/issues/668)to the detections tables, and anywhere else detections are shown on the website, would be helpful. Many people still assume that anything on the detections is equivalent to a bird sighted and photographed by multiple observers.

**Comment by adamdsmith** [comment]
This again seems related to runLen. All but one of these detections are runLen == 2

**Comment by jsayers-bsc** [comment]
Just looked at [Trunk Island Bermuda](https://motus.org/data/receiver/deployment/plot/?id=5520) and a good portion of those problematic detections are now gone...

**Comment by adamdsmith** [comment]
Following up on something @cjardine-bsc mentioned that now puzzles me as well---`pulseCount` should be, at a minimum, 4 * `numHits` since each hit is composed of 4 pulses. But this is often not the case because 

1. `pulseCount` is `NA` (perhaps related to some weird consequence of batch reruns as we discussed on a call recently), or
1. `pulseCount` is provided, but is inadequate to explain the `numHits`

I'm left to assume the `numHits` is the most consistent, and likely most relevant, summary of activity. It doesn't matter if a receiver is logging 200K pulses an hour if none of them string together as hits...

For an example, looking at the activity tables behind 5 large tagging projects spanning over 750 deviceIDs which, admittedly, I'm not sure what that is. Receivers?)):

- 1.5 M hourly activity records
- 79% as expected, `pulseCount`/4 >= `numHits`
- 15% `pulseCount`/4 < `numHits`, often exceedingly so
- 6% `NA` `pulseCount`s

Here's a sample of 150K activity records. Red points above the line have inadequate `pulseCount`s to explain `numHits`:

![image](https://user-images.githubusercontent.com/3195855/77217516-93f61380-6af9-11ea-9b90-ee1b18074eb1.png)

**Comment by jsayers-bsc** [comment]
A couple of the others I noticed today: 

[https://motus.org/data/receiverDeployment?id=3051](https://motus.org/data/receiverDeployment?id=3051)
[https://motus.org/data/receiverDeployment?id=5339](https://motus.org/data/receiverDeployment?id=5339
)

As with the others, many/most/all? of the false positives that show up on one of these "events" is runLen == 2, meaning they shouldn't be showing up in the detection tables in the first place.

Following through on some of the false detections will lead to yet more stations that experienced a similar event and "detected" many of the same tags. Perhaps tags that are particularly prone to being "detected" in the noise. Frequent fliers, expect they may not actually be going anywhere.

**Comment by StuMackenzie** [comment]
Closing in deference to #610 - requires followup.
